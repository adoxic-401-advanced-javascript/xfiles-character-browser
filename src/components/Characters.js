import React from 'react';
import PropTypes from 'prop-types';

const Characters = ({ characters }) => {
  const mappedCharacters = characters[0].map(character => {
    return (
      <li key={character._id}>
        <h3>{character.name}</h3>
        <img src={character.photoUrl} />
      </li>
    );
  });
 
  return (
    <ul>
      {mappedCharacters}
    </ul>
  );
};

Characters.propTypes = {
  characters: PropTypes.array
};

export default Characters;
