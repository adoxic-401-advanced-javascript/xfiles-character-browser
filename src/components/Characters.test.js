import React from 'react';
import { shallow } from 'enzyme';
import Characters from './Characters.js';

describe('Characters component', () => {
  it('renders Characters', () => {
    const wrapper = shallow(<Characters characters={[[{ _id: 'characters' }]]} />);
    expect(wrapper).toMatchSnapshot();
  });
});
