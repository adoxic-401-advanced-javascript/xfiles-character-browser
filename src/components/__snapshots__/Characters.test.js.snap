// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Characters component renders Characters 1`] = `
<ul>
  <li
    key="characters"
  >
    <h3 />
    <img />
  </li>
</ul>
`;
